mgc/discrim/base.py

```diff
@@ -0,0 +1,77 @@
+from abc import ABC, abstractmethod
+from sklearn.utils import check_X_y
+import numpy as np
+from .._utils import euclidean
+
+
+class DiscriminabilityTest(ABC):
+    r"""
+    A base class for a Discriminability test.
+    """
+
+    def __init__(self):
+        self.pvalue_ = None
```

Comment:
add `self.stat = None`

Reply:
For 2 sample test, we have two stat. So, cannot do this in the base class.